app/src/androidTest/java/ch/epfl/sweng/swenggolf/LeaderboardTest.java

```diff
@@ -0,0 +1,108 @@
+import ch.epfl.sweng.swenggolf.database.DatabaseUser;
+import ch.epfl.sweng.swenggolf.database.FakeDatabase;
+import ch.epfl.sweng.swenggolf.main.MainMenuActivity;
+import ch.epfl.sweng.swenggolf.profile.User;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.contrib.RecyclerViewActions.actionOnItem;
+import static android.support.test.espresso.matcher.ViewMatchers.hasDescendant;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withContentDescription;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static java.lang.Thread.sleep;
+import static junit.framework.TestCase.fail;
+
+
+@RunWith(AndroidJUnit4.class)
+public class LeaderboardTest {
+
+    @Rule
+    public final IntentsTestRule<MainMenuActivity> mActivityRule =
+            new IntentsTestRule<>(MainMenuActivity.class, false, false);
+
+    public static RecyclerViewMatcher withRecyclerView(final int recyclerViewId) {
+        return new RecyclerViewMatcher(recyclerViewId);
+    }
+
+    /**
+     * Set up a fake database with two offers.
+     */
+    protected static void setUpFakeDatabase() {
+        Database database = new FakeDatabase(true);
+        Config.goToTest();
+        User user1 = new User("username1", "userId1", "email1",
+                "photo1", "", "DEFAULT_DESCRIPTION1", 30);
+        User user2 = new User("username2", "userId2", "email2",
+                "photo2", "", "DEFAULT_DESCRIPTION2", 50);
+        database.write("users", "userId1", user1);
+        database.write("users", "userId2", user2);
+        Database.setDebugDatabase(database);
+        DatabaseUser.addUser(Config.getUser());
+    }
+
+    /**
+     * Configures a fake database and enables TestMode.
+     */
+    @Before
+    public void init() {
+        setUpFakeDatabase();
+        Config.goToTest();
+        mActivityRule.launchActivity(new Intent());
+        mActivityRule.getActivity().getSupportFragmentManager().beginTransaction()
+                .add(R.id.centralFragment, new Leaderboard()).commit();
+    }
+
+
+
+    @Test
+    public void brokenDatabase() {
+        Database database = new FakeDatabase(false);
+        Database.setDebugDatabase(database);
+    }
+
+    @Test
+    public void userCorrectlyDisplayedAfterClickOnList() {
+        try {
+            sleep(500);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        onView(withId(R.id.users_recycler_view)).perform(actionOnItem(hasDescendant(
+                ViewMatchers.withText(Leaderboard.userList.get(0).getUserName())), click()));
```

Comment:
Here you can use atPosition of the RecyclerViewMatcher.

Reply:
Since you access a specific user of your leaderboard (that should be in a specific order since it's the point of this leaderboard) you can do "(new RecyclerViewMatcher(R.id.users_recycler_view)).atPosition(x).perform(...)" where x should be 0 since the first element is the second user that has more points. It also allows you to have the user as a private attribute in the LeaderBoard Fragment.

Reply:
Ok capito